### PR TITLE
CO-1671 implement and test PATCH /ingredients/{ingredientId}

### PR DESCRIPTION
**Pull request by jemisonf**

### CODE REVIEW

File: src/utils/bind-params.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * @param {object} body
+ * @param {object} allowedProperties an object whose keys are the only keys
+ * allowed from `body.data` in `bindParams`
+ * @param {object} baseValue use to include other out bind variables that are not in body
+ * @returns {object} bindparams
+ * @throws {Error} when an attribute not in doughsProperties is found in body.
+ * It's assumed that body has already been checked for invalid properties meaning that when
+ * one is found here it's a server error rather than a user error.
+ */
+const getBindParams = (body, allowedProperties, baseValue = {}) => {
```
Comment (jemisonf) [inline src/utils/bind-params.js]: Moved this method to a util because it was getting used in both DAO classes

File: openapi.yaml [hunk 2]
```diff
@@ -786,7 +786,21 @@ definitions:
...
+                example: Sausage
+              ingredientType:
+                type: string
+                description: the type of the ingredient
+                enum: *ingredientTypeEnum
+                example: meat
+              notes:
+                type: string
+                description: additional information about using or preparing the ingredient
+                example: brown the sausage in a pan before baking
```
Comment (jemisonf) [inline openapi.yaml]: Same issue as in #17, needed to remove default values

File: src/api/v1/db/oracledb/ingredients-dao.js [hunk 3]
```diff
@@ -215,4 +216,55 @@ const getIngredientById = async (id) => {
...
+const createPatchQueryAndBindParams = (body) => {
+  const inBindParams = getBindParams(body, ingredientsProperties);
+
+
+  const query = ingredientsPatchQuery(inBindParams);
+
+  const bindParams = {
+    ...inBindParams,
+    ...ingredientsOutBindParams,
+    ...{ id: body.data.id },
```
[the remark this answers is not in the document]
Comment (jemisonf) [reply src/api/v1/db/oracledb/ingredients-dao.js]: looks like yes

File: openapi.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jemisonf) [reply openapi.yaml]: as in, adding `type`, `id`, and `attributes` as required?

File: openapi.yaml [hunk 4]
```diff
@@ -782,11 +782,27 @@ definitions:
   IngredientPatchBody:
     properties:
       data:
```
Comment (ianbrown9475) [inline openapi.yaml]: Do you also want to have `data` be required?

Comment (jemisonf) [reply openapi.yaml]: yep. should be fixed.

**Pull request merged**